clanker-deployer/DeployToken.ts

```diff
@@ -0,0 +1,646 @@
+import {
+  encodeFunctionData,
+  encodeAbiParameters,
+  encodePacked,
+  keccak256,
+  parseUnits,
+  createPublicClient,
+} from 'viem';
+import { Clanker_v3_1_abi } from './Clanker.js';
+import { createWalletClient, http } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+import { base } from 'viem/chains';
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+// Real value for the Wrapped ETH (WETH) address.
+const WETH_ADDRESS =
+  '0x4200000000000000000000000000000000000006' as `0x${string}`;
+const CLANKER_FACTORY_V3_1 =
+  '0x2A787b2362021cC3eEa3C24C4748a6cD5B687382' as `0x${string}`;
+
+// Type definitions
+interface IDeployFormData {
+  name: string;
+  symbol: string;
+  image?: File | null;
+  imageUrl?: string;
+  description?: string;
+  // Amount of paired token to use for the initial buy (creates initial liquidity)
+  devBuyAmount?: string | number;
+  // Percentage of token supply to lock in vesting contract (0-100)
+  lockupPercentage?: number;
+  // Unix timestamp when locked tokens can be released
+  vestingUnlockDate?: bigint | null;
+  // Whether to perform an initial buy with paired token (creates liquidity)
+  // This controls whether devBuyAmount is used
+  enableDevBuy?: boolean;
+  // Whether to enable token lockup/vesting
+  // This controls whether lockupPercentage and vestingUnlockDate are used
+  enableLockup?: boolean;
+  // Address to receive protocol fees (if applicable)
+  feeRecipient?: string | null;
+  telegramLink?: string;
+  websiteLink?: string;
+  xLink?: string;
+  marketCap?: string;
```

Comment:
same with this variable, is it used? i might not understand typescript

Reply:
they're used to build the metadata!